### ISSUE TITLE
jax.numpy.squeeze: error for nonexistent / nonsingleton axis

### ISSUE THREAD

**Issue by GeorgOstrovski** [google/jax#2284]
Numpy's squeeze() function fails with an error if the passed axis is either missing or has a size other than 1. jax.numpy.squeeze() should presumably have the same behaviour.

**Comment by hawkinsp** [comment]
Sound reasonable. Contributions welcome!

**Comment by mattjj** [comment]
Sounds like your #2470 fixed this. Thanks, @GeorgOstrovski !
